### ISSUE TITLE
Possibility to set "'Accept-Language: de" header

### ISSUE THREAD

**Issue by cb-Bhuvana** [gocardless/gocardless-pro-java#11]
Hi There,

What about a possibility to set the header mentioned above, or any other language?

I need this in order to get the mandate pdf in required languages based on the customer language.

https://developer.gocardless.com/api-reference/#helper-endpoints-mandate-pdfs

Thanks & Best,
Bhuvana

**Comment by timrogers** [comment]
@cb-Bhuvana At the moment, this isn't as easy to do in the library as it should be. There is a way, but it's a bit of a clunky one - we'll look at making this simpler in a future version.

* The [`HttpClient`](https://github.com/gocardless/gocardless-pro-java/blob/master/src/main/java/com/gocardless/http/HttpClient.java) has a `execute(ApiRequest<T> apiRequest, Map<String, String> headers)` method, which you can use to pass in headers
* When you call `client.mandatePdfs().create()`, you get back an instance which inherits from [`ApiRequest`](https://github.com/gocardless/gocardless-pro-java/blob/master/src/main/java/com/gocardless/http/ApiRequest.java), which has a `HttpClient getHttpClient()` method
* You can get this `HttpClient`, and then use its `execute` method to run your request with some custom headers, specifying the `Accept-Language` header

All in all, the solution would be something like this:

```java
import java.util.HashMap;
import java.util.Map;

import static com.gocardless.GoCardlessClient.Environment.SANDBOX;
import com.gocardless.services.MandatePdfService.MandatePdfCreateRequest;
import com.gocardless.http.HttpClient;

String accessToken = "your_access_token_here";
GoCardlessClient client = GoCardlessClient
    .newBuilder(accessToken)
    .withEnvironment(SANDBOX)
    .build();

MandatePdfCreateRequest request = client.mandatePdfs().create().withLinksMandate("MD123");
HttpClient httpClient = request.getHttpClient();

Map<String, String> headers = new HashMap<String, String>();
headers.put("Accept-Language", "de-DE");

MandatePdf mandatePdf = httpClient.execute(request, headers);
```

Can you let me know how you get on, and if this solves your problem? Thanks for leaving your issue!

**Comment by cb-Bhuvana** [comment]
@timrogers  Thanks for getting back quickly.


Can you check the comments I have added in the code?

`import java.util.HashMap;
import java.util.Map;

import static com.gocardless.GoCardlessClient.Environment.SANDBOX;
import com.gocardless.services.MandatePdfService.MandatePdfCreateRequest;
import com.gocardless.http.HttpClient;

String accessToken = "your_access_token_here";
GoCardlessClient client = GoCardlessClient
    .newBuilder(accessToken)
    .withEnvironment(SANDBOX)
    .build();

MandatePdfCreateRequest request = client.mandatePdfs().create().withLinksMandate("MD123");
HttpClient httpClient = request.getHttpClient(); //**getHttpClient** is default method, so not accessible.

Map<String, String> headers = new HashMap<String, String>();
headers.put("Accept-Language", "de-DE");

MandatePdf mandatePdf = httpClient.execute(request, headers); //**excute** is again default method, so not accessible
`

**Comment by timrogers** [comment]
Aha - I didn’t notice that! Sorry for giving you false hope.

Could you drop me an email at tim@gocardless.com? I’d like to look into
this further.

**Comment by timrogers** [comment]
This issue is fixed in `v3.3.0-rc1` and the final `v3.3.0` to be released today. It adds a `withHeader(String headerName, String headerValue)` to request objects that you can chain on, for example after the `withLinksMandate` in @cb-Bhuvana's example above.
